core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -312,25 +312,30 @@ public int hashCode() {
         return nodeId.hashCode();
     }
 
+    private String toString;
```

Comment:
> I would prefer not to calculate the toString all the time. I'm ok with the optimization...

I'm fine with that, I pushed cef99817be0367d4e2c4fc402a51dc4175c87955 to revert the eager evaluation.